game/lib/profile/completed_challenges_page.dart

```diff
@@ -89,8 +89,8 @@ class CompletedChallengesPage extends StatelessWidget {
                   var challenge = challengeModel.getChallengeById(challengeId);
                   if (challenge != null) {
                     pictureList.add(challenge.imageUrl!);
-                    locationList.add(
-                        friendlyLocation[challenge.location?.name ?? 'ANY']);
+                    locationList
```

Comment:
You have a null coalescing operator for location but not for imageUrl, doesn't make sense logically.